subsys/net/lib/sockets/Kconfig

```diff
@@ -139,6 +139,20 @@ config NET_SOCKETS_CAN_RECEIVERS
 	  The value tells how many sockets can receive data from same
 	  Socket-CAN interface.
 
+config NET_SOCKETPAIR
+	bool "Support for the socketpair syscall [EXPERIMENTAL]"
```

Comment:
As the code is calling `k_malloc()`, then we would need have CONFIG_HEAP_MEM_POOL_SIZE set.
So we should have here `depends on HEAP_MEM_POOL_SIZE != 0`